cardinal/ecs/world.go

```diff
@@ -562,14 +569,34 @@ func (w *World) StartGameLoop(ctx context.Context, loopInterval time.Duration) {
 		w.Logger.Warn().Msg("No systems registered.")
 	}
 	go func() {
+		w.isGameLoopRunning.Store(true)
+		wasEndGameLoopSignalReceived := false
 		for range time.Tick(loopInterval) {
+			if wasEndGameLoopSignalReceived {
+				break
+			}
 			if err := w.Tick(ctx); err != nil {
 				w.Logger.Panic().Err(err).Msg("Error running Tick in Game Loop.")
 			}
+			select {
+			case <-w.endGameLoopCh:
```

Comment:
Both time.Tick and w.endGameLoopCh are channels, and in this pattern you don't check for the end of the game loop until after a tick has been run. I'd suggest:
```
ticker := time.Tick(loopInterval)
for {
    select{
        case <-ticker:
            // Run the tick.
        case <-w.endGameLoopCh:
            // cleanup/break out. I like @technicallyty suggestion of using a label to break out of this nested context
    }
}
```

Reply:
Actually I think it can still occur even if it's placed after a tick. Yeah will discuss a bit during standup.

Reply:
As discussed in standup I will be following the select pattern above but also I'll need to check if the queue is empty before exiting this loop. If the queue is not empty tick it.